### PR TITLE
LPS-150420 Refactor tests on WebContentWithCustomStructure file | master 

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-150420

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplay.testcase [hunk 1]
```diff
@@ -799,6 +848,74 @@ Welcome to Liferay Community Edition Portal 7.4.0 CE GA1''';
...
+
+		WebContentDisplayPortlet.edit(
+			webContentTextSingleLine = "WC WebContent Text Edit",
+			webContentTitleEdit = "WC WebContent Title Edit");
+
+		WebContentDisplayPortlet.viewContent(
+			webContentContent = "WC WebContent Text Edit",
+			webContentTitle = "WC WebContent Title Edit");
+	}
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplay.testcase:918]: Keep in the original file

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind updating according to comments? Thanks

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind updating according to comments? Thanks

Hi @Tim-Cao, PR updated. Would you have a look at? Thanks!

**Comment by Tim-Cao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 2]
```diff
@@ -863,444 +826,379 @@ definition {
 		}
 	}
 
+	@description = "Add a web content based on new structure with a text field."
 	@priority = "4"
 	test AddWebContentWithStructure {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:831]: AddWebContentWithTextField

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 3]
```diff
@@ -1870,168 +1769,257 @@ definition {
...
+			AssertElementNotPresent(locator1 = "WC#SOURCE_GARBAGE_DATA");
+		}
+	}
 
-		WebContent.addCP(
-			webContentContent = "WC WebContent Content",
-			webContentTitle = "WC WebContent Title");
+	@description = "The user cannot add a web content, structure and template without a title."
+	@priority = "4"
+	test WebContentAdminNullTitleValidation {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:1978]: WebContentAdministrationDataValidation

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 4]
```diff
@@ -1870,168 +1769,257 @@ definition {
 		}
 	}
 
+	@description = "The user can view a global web content structure via add menu icon."
 	@priority = "3"
-	test ViewGlobalStructureViaAddMenuInLiferaySite {
-		property testray.component.names = "Web Content Administration";
+	test ViewGlobalStructureViaAddIcon {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:1774]: ViewGlobalStructureAvailableToNormalSite

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 5]
```diff
@@ -863,444 +826,379 @@ definition {
...
+				templateName = "Template Edit",
+				webContentTitle = "Web Content Title");
+		}
 	}
 
+	@description = "Can edit a web content from global structure in web content display widget."
 	@priority = "4"
-	test EditWebContentTemplate {
-		property testray.component.names = "Web Content Administration";
+	test EditWebContentFromGlobalStructure {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:1014]: EditWebContentWithGlobalStructure

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 6]
```diff
@@ -298,338 +321,278 @@ definition {
...
-		PortletEntry.save();
-
-		JSONUser.addUser(
-			userEmailAddress = "test1@liferay.com",
-			userFirstName = "test1",
-			userLastName = "test1",
-			userScreenName = "test1test");
+	@description = "The user can add a web content via Web Content Display widget based on global structure and local template."
+	@priority = "4"
+	test AddWebContentWithGlobalStructureAndLocalTemplate {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:465]: AddWebContentViaWCDWithGlobalStructureAndLocalTemplate

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 7]
```diff
@@ -24,271 +29,289 @@ definition {
...
-			nestedField = "Numeric",
-			nestedFieldData = "123");
+			AssertConsoleTextNotPresent(value1 = "freemarker.core.NonStringException");
+		}
+	}
 
-		WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+	@description = "Add a web content with custom structure that has nested fields."
+	@priority = "4"
+	test AddStructureWithSubfield {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:204]: AddWebContentWithNestedFields

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 8]
```diff
@@ -24,271 +29,289 @@ definition {
...
-		NavItem.gotoStructures();
-
-		WebContentStructures.addCP(
-			structureDescription = "WC Structure Description",
-			structureName = "WC Structure Name");
-
-		DataEngine.addField(
-			fieldFieldLabel = "Text",
-			fieldName = "Text");
+	test AddStructureWithSeparatorField {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:156]: AddWebContentWithSeparatorField

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase [hunk 9]
```diff
@@ -24,271 +29,289 @@ definition {
...
+		}
 	}
 
-	@description = "This is a use case for LPS-52248."
-	@priority = "4"
-	test AddStructureWithSeparatorField {
-		WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+	@description = "This is a use case for LPS-69560. Can add a web content with custom structure that has a geolocation field."
+	@priority = "5"
+	test AddStructureWithGeolocationField {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/structure/WebContentWithCustomStructures.testcase:94]: AddWebContentWithWebContentAndGeolocationFields

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind updating according to comments? Thanks

**Comment by Tim-Cao** [comment]
Good job! @ozysouza

**Comment by Tim-Cao** [comment]
See changes https://github.com/liferay-echo/liferay-portal/pull/8033/commits/76e985c6997a05c8305a5f0236514bf2485f5162
@ozysouza

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**